src/nomad_simulations/outputs.py

```diff
@@ -15,38 +15,153 @@
+        output property is calculated.
+        """,
+    )
+
+    is_derived = Quantity(
+        type=bool,
+        default=False,
+        description="""
+        Flag indicating whether the output property is derived from other output properties. We make
+        the distinction between directly parsed, derived, and post-processing output properties:
+            - Directly parsed: the output property is directly parsed from the simulation output files.
+            - Derived: the output property is derived from other output properties. No extra numerical settings
+                are required to calculate the output property.
+            - Post-processing: the output property is derived from other output properties. Extra numerical settings
+                are required to calculate the output property.
+        """,
+    )
+
+    outputs_ref = Quantity(
+        type=Reference(SectionProxy('BaseOutputs')),
+        description="""
+        Reference to the `BaseOutputs` section from which the output property was derived. This is only
+        relevant if `is_derived` is set to True.
+        """,
+    )
+
+    def check_is_derived(self, is_derived: bool, outputs_ref) -> Optional[bool]:
+        if not is_derived:
+            if outputs_ref is not None:
+                return True
+            return False
+        elif is_derived and outputs_ref is not None:
+            return True
+        return None
+
     def normalize(self, archive, logger) -> None:
         super().normalize(archive, logger)
-        self.logger = logger
+
+        check_derived = self.check_is_derived(self.is_derived, self.outputs_ref)
+        if check_derived is not None:
+            self.is_derived = check_derived
+        else:
+            logger.error(
+                'A derived output property must have a reference to another `Outputs` section.'
+            )
+            return
+
+
+class Outputs(BaseOutputs):
+    """
+    Output properties of a simulation.
+
+    # ! add more description once we defined the output properties
+    """
+
+    n_scf_steps = Quantity(
+        type=np.int32,
+        description="""
+        Number of self-consistent steps to converge the output property.
+        """,
+    )
+
+    scf_step = SubSection(
+        sub_section=BaseOutputs.m_def,
+        repeats=True,
+        description="""
+        Self-consistent (SCF) steps performed for converging a given output property.
+        """,
+    )
+
+    is_converged = Quantity(
+        type=bool,
+        description="""
+        Flag indicating whether the output property is converged or not. This quantity is connected
+        with `SelfConsistency` defined in the `numerical_settings.py` module.
+        """,
+    )
+
+    self_consistency_ref = Quantity(
+        type=SelfConsistency,
+        description="""
+        Reference to the `SelfConsistency` section that defines the numerical settings to converge the
+        output property.
+        """,
+    )
```

Comment:
Call it `is_scf_converged`.

I understand your approach here. Probably, it's clearer for people extending the schema to rigorously enforce the distinction _method_ and _output_.
Still, it feels kind of weird to have `is_converged` exist separately from its specification...